### DISCUSSION TITLE
Why randomly perturb the center when building the dataset instead of using a new pipeline to perturb the data as it is loaded?

### DISCUSSION THREAD

**Discussion by yinaoxiong** [open-mmlab/mmpose#1350] [Q&A] [answered]
I found that when the dataset was in training mode, the existing code would build the dataset with a random perturbation to the center of the image. As the following code shows
https://github.com/open-mmlab/mmpose/blob/41f36fe1315d986da9a88646b57b10b90e125555/mmpose/datasets/datasets/base/kpt_2d_sview_rgb_img_top_down_dataset.py#L144

What I find puzzling is that shouldn't this be the role that pipeline needs to play? Using a pipeline would better specify the probability of randomization and the range of offsets, as well as make the center of the same image different at different epochs.

Or maybe there's some special compatibility issue that requires keeping a piece of code like this

**Comment by ly015** [comment, marked as answer]
Yes, it is more reasonable to do the bbox perturbation in the pipeline. Actually, we have moved it to the pipeline in #1311 and it will be merged into the master branch in the next released version (v0.26.0).

Although as you mentioned, doing perturbation in the datasets leads to the issue that the same results are used for every epoch, it doesn't seem to harm the performance, as shown by our recent experiment. So you can safely use the model trained by the old version code.

**Reply by yinaoxiong** [reply]
> Yes, it is more reasonable to do the bbox perturbation in the pipeline. Actually, we have moved it to the pipeline in #1311 and it will be merged into the master branch in the next released version (v0.26.0).

I get it, thanks for the answers and the outstanding work 😁.
